src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState } from 'react';
+import { Todo, User } from '../../types';
+import usersFromServer from '../../api/users';
+
+type Props = {
+  onAdd: (newTodo: Todo) => void;
+  users: User[];
+};
+
+export const TodoForm: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState<number>(0);
+
+  const [title, setTitle] = useState<string>('');
+  const [titleError, setTitleError] = useState<boolean>(false);
+  const [userId, setUserId] = useState<number>(0);
+  const [userError, setUserError] = useState<boolean>(false);
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+    setTitleError(false);
+    setUserError(false);
+  };
+
+  const handleTitleUpdate = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitleError(false);
+    setTitle(event.target.value);
+  };
+
+  const handleUserUpdate = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserError(false);
+    setUserId(+event.target.value);
+  };
+
+  const handleAdd = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title) {
+      setTitleError(true);
+    }
+
+    if (!userId) {
+      setUserError(true);
+    }
+
+    if (!title || !userId) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      title,
+      userId: userId,
+      completed: false,
+      user: usersFromServer.find(user => user.id === userId),
```

Comment:
Ensure that the `user` property in the new todo is correctly handled. If `find` returns `undefined`, the `user` property will be `undefined`, which might not be the desired behavior. Consider handling this case appropriately.